src/components/molecules/NavSearchBar/NavSearchBar.tsx

```diff
@@ -79,72 +82,78 @@ const NavSearchBar = () => {
     <div className="nav-search-links">
       <div className="nav-search-icon" />
       <NavSearchBarInput value={searchQuery} onChange={setSearchQuery} />
+
       {isTruthy(searchQuery) && (
         <div className="nav-search-close-icon" onClick={clearSearchQuery} />
       )}
-      {isTruthy(numberOfSearchResults) && (
-        <>
-          <div className="nav-search-results">
-            <div className="nav-search-result-number">
-              <b>{numberOfSearchResults}</b> search results
-            </div>
-            {searchResult.rooms.map((room, index) => {
-              return (
+
+      {numberOfSearchResults > 0 && (
+        <div className="nav-search-results">
+          <div className="nav-search-result-number">
+            <b>{numberOfSearchResults}</b> search results
+          </div>
+
+          {/* @debt we really shouldn't be using the index as part of the key here, it's unstable.. but rooms don't have a unique identifier */}
+          {searchResult.rooms.map((room, index) => {
+            return (
+              <div
+                className="row"
+                key={`room-${room.title}-${index}`}
+                onClick={() => setSelectedRoom(room)}
+              >
                 <div
-                  className="row"
-                  key={`room-${index}`}
-                  onClick={() => setSelectedRoom(room)}
-                >
-                  <div
-                    className="result-avatar"
-                    style={{
-                      backgroundImage: `url(${room.image_url})`,
-                    }}
-                  ></div>
-                  <div className="result-info">
-                    <div className="result-title">{room.title}</div>
-                    <div>Room</div>
-                  </div>
+                  className="result-avatar"
+                  style={{
+                    backgroundImage: `url(${room.image_url})`,
+                  }}
+                />
+                <div className="result-info">
+                  <div className="result-title">{room.title}</div>
+                  <div>Room</div>
                 </div>
-              );
-            })}
-            {searchResult.events.map((event, index) => {
-              return (
-                <div className="row" key={`event-${index}`}>
-                  <div>
-                    <div>{event.name}</div>
-                    <div>Event</div>
-                  </div>
+              </div>
+            );
+          })}
+
+          {searchResult.events.map((event) => {
+            return (
+              <div className="row" key={`event-${event.id ?? event.name}`}>
```

Comment:
The typing says that `event.id` can apparently be `undefined` here, so I figured defaulting to `event.name` was a good fallback. 

Again, not a perfect solution, ideally we would just always use `event.id` as it would be a unique value.